app/gui/src/dashboard/layouts/Drive/Categories/Category.ts

```diff
@@ -202,64 +207,144 @@ export function areCategoriesEqual(a: Category, b: Category) {
 
 /** Whether an asset can be transferred between categories. */
 export function canTransferBetweenCategories(from: Category, to: Category) {
+  const operation = dropOperationBetweenCategories(from, to)
+
+  return operation !== 'cancel'
+}
+
+/**
+ * The drop operation to use when transferring assets between categories.
+ * @param from - The category to transfer from.
+ * @param to - The category to transfer to.
+ * @returns The drop operation to use.
+ */
+export function dropOperationBetweenCategories(from: Category, to: Category): DropOperation {
+  if (from.type === to.type) {
+    return 'cancel'
+  }
+
+  if (to.type === 'recent') {
+    return 'cancel'
+  }
+
+  if (isCloudCategory(from) || isCloudCategory(to)) {
+    if (isLocalCategory(from) || isLocalCategory(to)) {
+      return 'cancel'
```

Comment:
optional: for local -> cloud we have the "upload project" flow.

perhaps that should be opened as a separate issue though